src/views/Home/components/FarmStakingCard.tsx

```diff
@@ -38,20 +39,22 @@ const FarmedStakingCard = () => {
   const { account } = useWeb3React()
   const { t } = useTranslation()
   const farmsWithBalance = useFarmsWithBalance()
+  const masterChefContract = useMasterchef()
   const balancesWithValue = farmsWithBalance.filter((balanceType) => balanceType.balance.toNumber() > 0)
 
-  const { onReward } = useAllHarvest(balancesWithValue.map((farmWithBalance) => farmWithBalance.pid))
-
   const harvestAllFarms = useCallback(async () => {
     setPendingTx(true)
-    try {
-      await onReward()
-    } catch (error) {
-      // TODO: find a way to handle when the user rejects transaction or it fails
-    } finally {
-      setPendingTx(false)
+    // eslint-disable-next-line no-restricted-syntax
+    for (const farmWithBalance of balancesWithValue) {
+      try {
+        // eslint-disable-next-line no-await-in-loop
+        await harvest(masterChefContract, farmWithBalance.pid, account)
+      } catch (error) {
+        // TODO: find a way to handle when the user rejects transaction or it fails
+      }
```

Comment:
I think a cleaner way is to use `Promise.all`.

```
const harvestPromises = balancesWithValue.map((farmWithBalance) => {
 return harvest(masterChefContract, farmWithBalance.pid, account)
})
 
await Promise.all(harvestPromises)
 ```

Reply:
Nope, that approach doesn't show the next transaction window. It gets stuck at the first one